core/evidence.c

```diff
@@ -0,0 +1,114 @@
+/* Copyright (c) 2022 Intel Corporation
+ * Copyright (c) 2020-2022 Alibaba Cloud
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <librats/err.h>
+#include <librats/log.h>
+#include <librats/evidence.h>
+
+size_t _size_of_evidence(const char *type)
+{
+	if (!strcmp(type, "sgx_epid")) {
+		return sizeof(attestation_verification_report_t);
+	} else if (!strcmp(type, "sgx_ecdsa")) {
+		return sizeof(ecdsa_attestation_evidence_t);
+	} else if (!strcmp(type, "tdx_ecdsa")) {
+		return sizeof(tdx_attestation_evidence_t);
+	} else if (!strcmp(type, "sgx_la")) {
+		return sizeof(la_attestation_evidence_t);
+	} else if (!strcmp(type, "sev_snp")) {
+		return sizeof(snp_attestation_evidence_t);
+	} else if (!strcmp(type, "sev")) {
+		return sizeof(sev_attestation_evidence_t);
+	} else if (!strcmp(type, "csv")) {
+		return sizeof(csv_attestation_evidence_t);
+	}
+
+	RATS_WARN("Unhandled evidence type %s\n", type);
+	return -1;
+}
+
+rats_err_t serialize_evidence(const attestation_evidence_t *evidence, uint8_t **buffer,
+			      size_t *buffer_size)
```

Comment:
Is it better not to change the offset of `*buffer` and calculate the offset of `custom_claims_buffer` outside the function?

Reply:
Fixed in line 2 of the list. https://github.com/inclavare-containers/librats/pull/70#issuecomment-1296577806